set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -512,20 +512,21 @@ private int validateLastPage(long totalInCollection, int pageSize, int pageNr) t
 	String searchProfile = searchQuery.getSearchProfile();
 	long totalInCollection = results.getResultSize();
 
-	int lastPage = getLastPage(totalInCollection, pageSize);
+	int lastPage = validateLastPage(totalInCollection, pageSize, currentPage);
+	// get profile for pagination urls and item Page
+	// do not want to change the parameters of other methods used by other functionality
+	LdProfiles profile = null;
+	for (LdProfiles ldProfile : profiles) {
+		if (!ldProfile.equals(LdProfiles.FACETS)) {
```

Comment:
the equals method should be allways called on the constants (LdProfiles.FACETS)

Reply:
I assume we cannot have the debug profile in the list at this stage

Reply:
it would be good to move the selection of the profile to own method

Reply:
done